pyincore/analyses/mlenabledcgeslc/mlcgeslc.py

```diff
@@ -0,0 +1,178 @@
+from pyincore.utils import parse_files
+
+logger = pyglobals.LOGGER
+
+
+class MlEnabledCgeSlc(CoreCGEML):
+
+    model = "Machine Learning Enabled Computable General Equilibrium - Salt Lake City "
+
+    #### Coefficients files
+    DDS_coefficients_file = "DDS_coefficients.csv"
+    DY_coefficients_file = "DY_coefficients.csv"
+    MIGT_coefficients_file = "MIGT_coefficients.csv"
+    DFFD_coefficients_file = "DFFD_coefficients.csv"
+
+    #### Base value files
+    DS_base_val_file = "DS_base_val.csv"
+    GI_base_val_file = "GI_base_val.csv"
+    HH_base_val_file = "HH_base_val.csv"
+    FD_base_val_file = "FD_base_val.csv"
+
+    Base_KAP_file = "baseKAP.csv"
+
+    base_file_path = os.path.join(
+        pyglobals.PYINCORE_PACKAGE_HOME,
+        "analyses",
+        "mlenabledcgeslc",
+    )
+
+    model_filenames = {
+        "ds": os.path.join(
+            base_file_path,
+            DDS_coefficients_file,
+        ),
+        "dy": os.path.join(
+            base_file_path,
+            DY_coefficients_file,
+        ),
+        "migt": os.path.join(
+            base_file_path,
+            MIGT_coefficients_file,
+        ),
+        "dffd": os.path.join(
+            base_file_path,
+            DFFD_coefficients_file,
+        ),
+    }
+
+    filenames = [
+        os.path.join(
+            base_file_path,
+            DS_base_val_file,
+        ),
+        os.path.join(
+            base_file_path,
+            GI_base_val_file,
+        ),
+        os.path.join(
+            base_file_path,
+            HH_base_val_file,
+        ),
+        os.path.join(
+            base_file_path,
+            FD_base_val_file,
+        ),
+        os.path.join(
+            base_file_path,
+            Base_KAP_file,
+        ),
+    ]
+
+    def __init__(self, incore_client: IncoreClient):
+        sectors, base_cap_factors, base_cap, model_coeffs, cap_shock_sectors = (
+            parse_files(self.model_filenames, self.filenames)
+        )
+        self.base_cap_factors = base_cap_factors
+        self.base_cap = base_cap
+        self.model_coeffs = model_coeffs
+        self.cap_shock_sectors = cap_shock_sectors
+        super(MlEnabledCgeSlc, self).__init__(
+            incore_client, sectors, labor_groups=[f"L{gp}" for gp in range(1, 5)]
+        )  # 4 labor groups
+
+    def run(self) -> bool:
+        logger.info(f"Running {self.model} model...")
+        sector_shocks = pd.read_csv(
+            self.get_input_dataset("sector_shocks").get_file_path("csv")
+        )
+        # arrange the capital shocks in the same order as the sectors
+        shocks = []
+
+        for sector in self.cap_shock_sectors:
+            if sector.upper() not in [v.upper() for v in sector_shocks["sector"]]:
+                raise ValueError(
+                    f"Sector {sector} not found in the sector shocks file with\n {sector_shocks['sector']} sectors. \nPlease make sure you have used the correct capital shocks"
```

Comment:
Here too about the line length

Reply:
I think we would need to add precommit here as well so for now I have addresses these. We can later look into adding a consistent formatting across developers later.